packages/simcore-sdk/src/simcore_sdk/node_ports/filemanager.py

```diff
@@ -17,6 +19,27 @@
 CHUNK_SIZE = 1*1024*1024
 
 
+class ClientSessionContextManager:
+    #
+    # NOTE: creating a session at every call is inneficient and a persistent session
+    # per app is recommended.
+    # This package has no app so session is passed as optional arguments
+    # See https://github.com/ITISFoundation/osparc-simcore/issues/1098
+    #
+    def __init__(self, session=None):
+        self.active_session = session or ClientSession()
+        self.is_owned = self.active_session is not session
+
+    async def __aenter__(self):
+        return self.active_session
+
+    async def __aexit__(self, exc_type, exc, tb):
+        if self.is_owned:
+            warnings.warn("Optional session will be deprecated, pass instead controled session (e.g. from app[APP_CLIENT_SESSION_KEY])",
+                category=DeprecationWarning)
+            await self.active_session.close()
+
+

```

Comment:
very nice!

Reply:
As you see, in order to keep backwards compatibility, the session in the node-ports is optional. Therefore is responsibility of the client application to provide a unique session